src/crimson/osd/state.h

```diff
@@ -43,32 +51,47 @@ class OSDState {
                        : wait_for_active.get_shared_future();
   };
   bool is_prestop() const {
+    ceph_assert(seastar::this_shard_id() == PRIMARY_CORE);
     return state == State::PRESTOP;
   }
   bool is_stopping() const {
     return state == State::STOPPING;
   }
   bool is_waiting_for_healthy() const {
+    ceph_assert(seastar::this_shard_id() == PRIMARY_CORE);
     return state == State::WAITING_FOR_HEALTHY;
   }
   void set_preboot() {
+    ceph_assert(seastar::this_shard_id() == PRIMARY_CORE);
     state = State::PREBOOT;
   }
   void set_booting() {
+    ceph_assert(seastar::this_shard_id() == PRIMARY_CORE);
     state = State::BOOTING;
   }
   void set_active() {
     state = State::ACTIVE;
     wait_for_active.set_value();
     wait_for_active = {};
+    if (seastar::this_shard_id() == PRIMARY_CORE) {
+      std::ignore = container().invoke_on_others([](auto& osd_state) {
+        osd_state.set_active();
+      });
```

Comment:
I don't think it makes sense for set_active() to return before the updates on all cores have completed.  It should probably return a seastar::future<>.